modules/transcription-service-google-speech-impl/src/main/java/org/opencastproject/transcription/googlespeech/GoogleSpeechTranscriptionService.java

```diff
@@ -1043,8 +1077,13 @@ private String startWorkflow(String mpId, String wfDefId, Map<String, String> pa
     final AQueryBuilder q = assetManager.createQuery();
     final AResult r = q.select(q.snapshot()).where(q.mediaPackageId(mpId).and(q.version().isLatest())).run();
     if (r.getSize() == 0) {
-      // Media package not archived yet.
-      logger.warn("Media package {} has not been archived yet.", mpId);
+      if (!hasTranscriptionRequestExpired(jobId)) {
+        // Media package not archived but still within completion time? Skip until next time.
+        logger.warn("Media package {} has not been archived yet. Skipped.", mpId);
+      } else {
+        // Close transcription job and email admin
+        cancelTranscription(jobId, "Transcription job canceled, archived media package not found");
+      }
       return null;
     }
```

Comment:
It's too bad that the workflow service is not (or cannot be?) queried here to verify that the mpId still exists when the asset manager has no knowledge of the mpId.  For example, when the event is deleted before Google transcript is attached, a 5 hour timeout plus length of video plus buffer has to be waited through before cancelling the transcript job. .... it might be useful to include the countdown in that Skipped log line. For example 
```
logger.warn("Media package {} has not been archived yet. Will keep trying for {} more minutes before cancelling transcript job {}.", mpId, getRemainingTranscriptionExpireTimeInMin(jobId), jobId);
```

Reply:
As a future feature, it might be worth adding a listener for deleted events to the transcript service so it can cancel out jobs when the event is deleted before the transcript is attached.

Reply:
I like the idea of count down, I'll try to add it.
I'll add the second part of your comment in future improvements.

Reply:
Count down now added